CMakeLists.txt

```diff
@@ -1964,21 +1964,27 @@ endif()
 
 ###      COOLPROP TESTING APP       ###
 if(COOLPROP_CATCH_MODULE)
+
+  add_subdirectory("externals/Catch2")
   enable_testing()
 
-  list(APPEND APP_SOURCES "${CMAKE_CURRENT_SOURCE_DIR}/src/Tests/test_main.cxx")
   list(APPEND APP_SOURCES
        "${CMAKE_CURRENT_SOURCE_DIR}/src/Tests/CoolProp-Tests.cpp")
 
   # CATCH TEST, compile everything with catch and set test entry point
   add_executable(CatchTestRunner ${APP_SOURCES})
-  add_dependencies(CatchTestRunner generate_headers)
-  set_target_properties(
-    CatchTestRunner PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -DENABLE_CATCH")
+
+  target_link_libraries(CatchTestRunner PRIVATE Catch2::Catch2WithMain)
+  target_compile_definitions(CatchTestRunner PRIVATE ENABLE_CATCH)
   if(UNIX)
     target_link_libraries(CatchTestRunner ${CMAKE_DL_LIBS})
   endif()
-  add_test(ProcedureTests CatchTestRunner)
+
+  target_link_libraries(CatchTestRunner PRIVATE ${COOLPROP_LIBRARY_NAME})
```

Comment:
Will need to remove this as it's not needed since I now use APP_SOURCES instead of just using the files with tests in (per @ibell 's request/taste)